apm-sniffer/apm-toolkit-activation/apm-toolkit-trace-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/trace/ActiveSpanErrorInterceptor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+
+package org.apache.skywalking.apm.toolkit.activation.trace;
+
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.StaticMethodsAroundInterceptor;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author caoyixiong
+ */
+public class ActiveSpanErrorInterceptor implements StaticMethodsAroundInterceptor {
+    @Override public void beforeMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes,
+                                       MethodInterceptResult result) {
+        AbstractSpan activeSpan = null;
+        try {
+            activeSpan = ContextManager.activeSpan();
+            activeSpan.errorOccurred();
+            if (allArguments != null) {
+                if (allArguments.length == 1) {
+                    if (allArguments[0] != null && allArguments[0] instanceof Throwable) {
+                        activeSpan.log((Throwable) allArguments[0]);
+                    }
+                } else if (allArguments.length == 2) {
+                    if (allArguments[0] != null
+                            && allArguments[0] instanceof String
+                            && allArguments[1] != null
+                            && allArguments[1] instanceof String) {
```

Comment:
Too many `instanceof`s, which will slow the system. Please separate the interceptor and make the logic clear.

Reply:
ok